src/v/archival/ntp_archiver_service.cc

```diff
@@ -1282,7 +1284,8 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
     // uploaded.
     std::vector<ss::future<cloud_storage::upload_result>> all_uploads;
 
-    all_uploads.emplace_back(upload_segment(upload, std::move(locks)));
+    all_uploads.emplace_back(
+      upload_segment(upload_ctx.archiver_term, upload, std::move(locks)));
```

Comment:
nit: I think it'd make conceptual sense to add the archiver term to the `upload_candidate` instead of the `upload_ctx`, given most of the call-site updates include the candidate.

Reply:
The upload candidate comes from the `archival_policy` which doesn't know anything about the partition state or `ntp_archiver`. So, in order to do this we will have to pass archiver term to the policy or modify `upload_candidate` after it's generated by the policy. Changing call sites looks a bit less invasive IMO.